examples/decomposition/plot_incremental_pca.py

```diff
@@ -33,15 +33,27 @@
 y = iris.target
 
 n_components = 2
-ipca = IncrementalPCA(n_components=n_components, batch_size=10)
-X_ipca = ipca.fit_transform(X)
+X_ipca = IncrementalPCA(n_components=n_components)
+
+# Process the dataset in batches:
+batch = []
+for index in range(len(X)):
+    row = X[index]
+    batch.append(row)
+    if len(batch) == 10:
+        X_ipca.partial_fit(batch)
+        batch = []
```

Comment:
It would be more efficient to size batches:

```python
for idx in range(0, len(X), batch_size):
    X_batch = X[idx:idx + batch_size]
    ipca.partial_fit(X_batch)

...
```

Reply:
Yes, one could do that. I was trying to make the example rely less on numpy idioms and be readable by the masses for more general cases. :)

Reply:
Yes, that makes sense!